### ISSUE TITLE
gRPC return a success when HTTP 404?

### ISSUE THREAD

**Issue by anticpp** [grpc/grpc-go#1363]
Please answer these questions before submitting your issue.

### What version of gRPC are you using?
v1.2.0

### What version of Go are you using (`go version`)?
go version go1.7.4 linux/amd64

### What operating system (Linux, Windows, …) and version?
linux/CentOS Linux release 7.3.1611 7

### What did you do?
I have deployed an [envoy](https://lyft.github.io/envoy/) beside my client, which making gRPC calls from the envoy egress address. I just kill the upstream gRPC server, then the envoy can't proxy the requests, It just respond an HTTP/404 error. 

As envoy access logging:
```
[2017-07-10T12:36:49.793Z] "POST /xlsoa.core.certificate/SyncKey HTTP/2" 404 NR 0 0 0 - "-" "grpc-go/1.0" "ab98a660-7e03-906e-be83-03e24e9b0011" "127.0.0.1" "-"
```

### What did you expect to see?
Well, some gRPC error code is in my expectations.

### What did you see instead?
But It appears a success gRPC call, along with an 'empty' response message, as codes below:

```
if resp, err = c.SyncKey(context.Background(), req); err != nil {
    return nil, 0, errors.Wrap(err, "SyncKey fail")
}
```

**Comment by anticpp** [comment]
Well, It seems fix at gRPC 1.4.1?

```
2017/07/10 20:52:21 [xlsoa] [OauthKeySource] [Error] sync() error: SyncKey fail: rpc error: code = Unimplemented desc = Not Found
```

**Comment by menghanl** [comment]
This was fixed by #1195, which is included in release 1.4.0.

So right, it's fixed in 1.4.1.
